### ISSUE TITLE
[bug] Regression in bazel generator, 1.54.0 adds again build dependencies into bazel files

### ISSUE THREAD

**Issue by a4z** [conan-io/conan#12677]
Conan 1.54.0 re-introduce a problem for the bazel generator by adding build dependencies as runtime dependencies in bazel files.

A new version of [#12236](https://github.com/conan-io/conan/issues/12236) 

### Environment Details (include every applicable attribute)
  * Operating System+version: MacOS 
  * Compiler+version: Not relevant
  * Conan version: 1.54.0
  * Python version: 3.8.x , 3.10.x

### Steps to reproduce (Include if Applicable)

Use sqlcypher , with bazel generator. 

With Conan 1.53.0, the generated bazel file is correct

```

load("@rules_cc//cc:defs.bzl", "cc_import", "cc_library")


cc_import(
    name = "sqlcipher_precompiled",
    static_library = "lib/libsqlcipher.a",
)




cc_library(
    name = "sqlcipher",
    
    hdrs = glob(["include/**", "include/sqlcipher/**"]),
    
    
    includes = ["include", "include/sqlcipher"],
    
    
    defines = ["SQLITE_HAS_CODEC", "SQLITE_TEMP_STORE=2", "SQLCIPHER_CRYPTO_OPENSSL"],
    
    
    visibility = ["//visibility:public"],
    
    deps = [
        # do not sort
    
    ":sqlcipher_precompiled",
    
    
    
    "@openssl",
    
    ],
    
)
```

With Conan 1.54.0, the generated bazel file is like this

```

load("@rules_cc//cc:defs.bzl", "cc_import", "cc_library")


cc_import(
    name = "sqlcipher_precompiled",
    static_library = "lib/libsqlcipher.a",
)




cc_library(
    name = "sqlcipher",
    
    hdrs = glob(["include/**", "include/sqlcipher/**"]),
    
    
    includes = ["include", "include/sqlcipher"],
    
    
    defines = ["SQLITE_HAS_CODEC", "SQLITE_TEMP_STORE=2", "SQLCIPHER_CRYPTO_OPENSSL"],
    
    
    visibility = ["//visibility:public"],
    
    deps = [
        # do not sort
    
    ":sqlcipher_precompiled",
    
    
    
    "@openssl",
    
    "@tcl",
    
    "@gnu-config",
    
    "@zlib",
    
    ],
    
)

```

These 3,  tcl,  gnu-config,  zlib , are build dependencies and should not be listed. 

This looks like a problem that was fixed in [#12236](https://github.com/conan-io/conan/issues/12236)

**Comment by memsharded** [comment]
Hi @a4z 

I am having a look at this, but it doesn't seem a regression, but it never got properly fixed, as I see in ``BazelDeps``, dating back to 2021:

```python
2639071e10 (fanStefan                            2021-12-27 12:24:55 +0100  21)         for build_dependency in self._conanfile.dependencies.direct_build.values():
```

This should probably be made opt-in, as in other generators, lets try to implement it correctly for next release. Thanks very much for reporting it.

**Comment by memsharded** [comment]
I have been investigating this a bit, and apparently it is good. The ``build_requires`` are only added as binaries, not libraries, so it should be good.

It seems the fix didn't really landed in 1.53 or 1.54, but https://github.com/conan-io/conan/pull/12543 landed in 1.55.
Please update to 1.55 and let us know, thanks!

**Comment by a4z** [comment]
No it's not good! 
This works in a toy local project but totally breaks real-world usage in more complex scenarios where these dependencies do not exist on the build machines, only the runtime ones, but bazel looks for them when they are in the file.

**Comment by a4z** [comment]
but 1.55.0 seems to work and not add the build dependencies, 
will do some more testing, but thanks for the moment!

**Comment by a4z** [comment]
Yes. Looks good!
(had my own fix for 1.53 but somehow forgot about it .... details, sometimes to many ...  :-) )
